lib/browser/objects-registry.js

```diff
@@ -47,10 +47,12 @@ class ObjectsRegistry {
   // Dereference an object according to its ID.
   remove(webContentsId, id) {
     // Dereference from the storage.
-    this.dereference(id);
+    const noReferences = this.dereference(id);
 
-    // Also remove the reference in owner.
-    this.owners[webContentsId].delete(id);
+    if (noReferences) {
```

Comment:
Perhaps the naming should be changed? I think that the way that it is currently done is the best possible route.
This function is (eventually) called after a garbage collect happens on a remote reference. How should an external user of `objects-registry` know if garbage-collect #3 removed the last-known reference to the object?

(IMHO) the answer is that `objects-registry` maintains the logic of how many references there are the the remote object, just like it used to.